hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3398,7 +3398,6 @@ private void scan(HBaseRpcController controller, ScanRequest request, RegionScan
             }
             boolean mayHaveMoreCellsInRow = scannerContext.mayHaveMoreCellsInRow();
             Result r = Result.create(values, null, stale, mayHaveMoreCellsInRow);
-            lastBlock.setValue(addSize(rpcCall, r, lastBlock.getValue()));
```

Comment:
We do not need to count lastBlock any more?

Reply:
This call to `addSize` here is no longer necessary for the underlying scan, because now we track blocks in the ScannerContext.  This addSize call increments the `rpcCall` response cell size and block size based on the estimated block sizes. The new method is more accurate, and a few lines below we call `rpcCall.incrementResponseBlockSize` and `rpcCall.incrementResponseCellSize` to ensure the same rpc call accounting.

Reply:
I removed `lastBlock` from this method, but we still need it in the caller. I explain the reasonings in my response to your other comment.